lib/Macros/Sources/ObservationMacros/CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+#===--- CMakeLists.txt - Observation macros library ----------------------===#
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2023 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+#===----------------------------------------------------------------------===#
+
+if (SWIFT_SWIFT_PARSER)
+  add_library(ObservationMacros SHARED
+    ObservableMacro.swift)
+
+  set_target_properties(ObservationMacros
+    PROPERTIES
+      ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib/swift/host/plugins"
+      LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib/swift/host/plugins"
+  )
+
+  target_compile_options(ObservationMacros PRIVATE
+    $<$<COMPILE_LANGUAGE:Swift>:-runtime-compatibility-version>
+    $<$<COMPILE_LANGUAGE:Swift>:none>)
+
+  # Set the appropriate target triple.
+  if(SWIFT_HOST_VARIANT_SDK IN_LIST SWIFT_DARWIN_PLATFORMS)
+    set(DEPLOYMENT_VERSION "${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_DEPLOYMENT_VERSION}")
+  endif()
+
+  if(SWIFT_HOST_VARIANT_SDK STREQUAL ANDROID)
+    set(DEPLOYMENT_VERSION ${SWIFT_ANDROID_API_LEVEL})
+  endif()
+
+  get_target_triple(target target_variant "${SWIFT_HOST_VARIANT_SDK}" "${SWIFT_HOST_VARIANT_ARCH}"
+    MACCATALYST_BUILD_FLAVOR ""
+    DEPLOYMENT_VERSION "${DEPLOYMENT_VERSION}")
+
+    target_compile_options(ObservationMacros PRIVATE $<$<COMPILE_LANGUAGE:Swift>:-target;${target}>)
+ 
+  # Workaround a cmake bug, see the corresponding function in swift-syntax
+  function(force_target_macros_link_libraries TARGET)
+    cmake_parse_arguments(ARGS "" "" "PUBLIC" ${ARGN})
+
+    foreach(DEPENDENCY ${ARGS_PUBLIC})
+      target_link_libraries(${TARGET} PRIVATE
+        ${DEPENDENCY}
+      )
+      add_dependencies(${TARGET} ${DEPENDENCY})
+
+      add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
+        COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
+        DEPENDS ${DEPENDENCY}
+        )
+      target_sources(${TARGET} PRIVATE
+        ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
+      )
+    endforeach()
+  endfunction()
+
+  set(SWIFT_SYNTAX_MODULES
+    SwiftSyntax
+    SwiftSyntaxMacros
+  )
+
+  # Compute the list of SwiftSyntax targets
+  list(TRANSFORM SWIFT_SYNTAX_MODULES PREPEND "SwiftSyntax::"
+       OUTPUT_VARIABLE SWIFT_SYNTAX_TARGETS)
+
+  # TODO: Change to target_link_libraries when cmake is fixed
+  force_target_link_libraries(ObservationMacros PUBLIC
+    ${SWIFT_SYNTAX_TARGETS}
+  )
+
+  set(SWIFT_SYNTAX_LIBRARIES_SOURCE_DIR
```

Comment:
This is a big pile of CMake that it would be really nice to share with ASTGen and any other macro implementations we might land. That refactor doesn't have to happen with your PR necessarily.

Reply:
yea I was tempted to generalize this into a macro library specific function - but my rush was to get things in first and then have follow-on tasks (where we have more time) to clean that up.